MCH: improvements to the pedestal calibrator

aferrero2707 opened the pull request:
The pedestal processing code is modified to run in multi-threaded mode, to improve the processing speed on the calibrator node. The periodic logging of pedestals statistics is also improved, to better monitor and debug the pedestals data taking.

The multi-threading is needed because the calibrator currently runs at 100% CPU, and we think this is increasing unnecessarily the duration of the calibration runs and limiting the statistics that can be collected in a reasonable time.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 399a65e2f5c5d81e07d6bf5a08725cb3c3b3b6be at 2025-08-09 13:17:
```
++ cp /sw/slc9_x86-64/O2/14564-slc9_x86-64-local1/compile_commands.json .
+++ python3 -c 'import json, os; print(os.path.commonpath([x["file"] for x in json.loads(open("compile_commands.json").read()) if "sw/BUILD" not in x["file"] and "G__" not in x["file"] and x["file"].endswith(".cxx")]))'
++ O2_SRC=/sw/SOURCES/O2/14564-slc9_x86-64/0
++ [[ -e /sw/SOURCES/O2/14564-slc9_x86-64/0/CMakeLists.txt ]]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14564/399a65e2f5c5d81e07d6bf5a08725cb3c3b3b6be/build_O2_fullCI_slc9/pretty.html).

aferrero2707 committed:
[MCH] improvements to the pedestal calibrator

The pedestal processing code is modified to run in multi-threaded
mode, to improve the processing speed on the calibrator node

The periodic logging of pedestals statistics is also improved,
to better monitor and debug the pedestals data taking.

aferrero2707 commented:
@shahor02 @ehellbar could you please review this PR, and possibly include it in the next SW release for P2? The changes are expected to improve the speed and reliability of the MCH calibrations. Thanks!

It also requires https://github.com/AliceO2Group/O2DPG/pull/2089 for some related changes in the options.

shahor02 reviewed (approved):
looks fine, though you could probably avoid being blocked by mutexes by collecting the result of different threads to dedicated containers then merging them.
Anyway, this can be done later if needed.
